#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_new_functions.dpatch by Anthony Fok <foka@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Experimental new functions needed to replace libft-perl for defoma.
## DP: Unfinished.

@DPATCH@
diff -urNad libfont-freetype-perl-0.03~/FreeType.xs libfont-freetype-perl-0.03/FreeType.xs
--- libfont-freetype-perl-0.03~/FreeType.xs	2008-07-27 15:23:57.000000000 +0800
+++ libfont-freetype-perl-0.03/FreeType.xs	2008-07-27 15:23:57.000000000 +0800
@@ -22,6 +22,7 @@
 #include FT_OUTLINE_H
 #include FT_BBOX_H
 #include FT_TYPE1_TABLES_H
+#include FT_SFNT_NAMES_H
 
 #undef assert
 #include <assert.h>
@@ -765,6 +766,34 @@
         }
 
 
+long
+qefft2_face_number_of_charmaps (Font_FreeType_Face face)
+    CODE:
+        RETVAL = face->num_charmaps;
+    OUTPUT:
+        RETVAL
+
+int
+qefft2_face_sfnt_name_count (Font_FreeType_Face face)
+    CODE:
+        RETVAL = FT_Get_Sfnt_Name_Count(face);
+    OUTPUT:
+        RETVAL
+
+SV *
+qefft2_face_sfnt_name (Font_FreeType_Face face, FT_UInt idx)
+    PREINIT:
+        const char *ps_name;
+    CODE:
+        ps_name = FT_Get_Postscript_Name(face);
+        if (ps_name)
+            RETVAL = newSVpv(ps_name, 0);
+        else
+            RETVAL = &PL_sv_undef;
+    OUTPUT:
+        RETVAL
+
+
 MODULE = Font::FreeType   PACKAGE = Font::FreeType::Glyph   PREFIX = qefft2_glyph_
 
 
